src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java

```diff
@@ -258,6 +258,16 @@ private static void defineConnectionConfigs(ConfigDef result) {
                 connectionGroupOrder++,
                 Width.MEDIUM,
                 KUSTO_URL_DISPLAY)
+            .define(
+                KUSTO_ENGINE_URL_CONF,
+                Type.STRING,
+                "", // TODO [yischoen 2020-09-29]: Next major version bump, set to ConfigDef.NO_DEFAULT_VALUE to make it a required setting
```

Comment:
Why not require it? It seems to me that by guessing we're asking for trouble in the future. Sure, if a user has trouble with this they can always provide this as an optional parameter, but that would be extra troubleshooting and likely noise for us.

Adding @vladikbr and @anagha-microsoft to the discussion.

Reply:
Any input @vladikbr and @anagha-microsoft ?

Reply:
If we do not make this required, IMHO, it will make the whole setup more confusing (one mandatory, one optional).
I think we should bump the major version and make this a pair of mandatory params - KUSTO_QUERY_URI and KUSTO_INGEST_URI.

And align everything to "URI" all over.

Reply:
Regardless of deployment type (private vnet injected or not), on start up of a connector, we validate access to the database by doing a count (can be improved to a more lightweight operation).  For this we need the engine URI as well.

Strategic solution, IMO - 
1. Add the engine URL as a sink property - kusto.uri.query (engine is not that clear)
2. Rename existing ingest URL sink property to - kusto.uri.ingestion

While we do this, can we modify the check for valid access from table | count to something lightweight as well?
Further, Grab has reported issues with the connector 1.0.1 regarding the file being written to.  We can release all of these together.

Reply:
@vladikbr - why URI? Sure, all Kusto endpoints are URIs, but more specifically, they're also URLs.

@anagha-microsoft 
May we use kusto.query.uri and kusto.ingestion.uri instead? I think it's better to go from more general to more specific.

I just tested and confirmed the lightweight solution works, and will push that.

Reply:
We'll also need to update the URL config's name + add the engine URL config in about 10 places in the azure-kusto-labs\kafka-integration project.

Reply:
> @vladikbr - why URI? Sure, all Kusto endpoints are URIs, but more specifically, they're also URLs.
> 
> @anagha-microsoft
> May we use kusto.query.uri and kusto.ingestion.uri instead? I think it's better to go from more general to more specific.
> 
> I just tested and confirmed the lightweight solution works, and will push that.

Sure.  And happy to test when its ready.  Please let me know, thanks.

Reply:
> We'll also need to update the URL config's name + add the engine URL config in about 10 places in the azure-kusto-labs\kafka-integration project.

Yep.  Loads of places.

Reply:
This PR is ready if you wanted to run any additional tests on it.
Would you please also give me permissions for https://github.com/Azure/azure-kusto-labs?

Reply:
Made you admin to the repo.  Thanks.  Do let me know when you have a jar uploaded to "releases", will test it out and share results.